doc/releases/release-notes-3.2.rst

```diff
@@ -123,6 +123,12 @@ Bluetooth
 
 * Host
 
+  * Added a new callback :c:func:`rpa_expired` in the struct :c:struct:`bt_le_ext_adv_cb`
```

Comment:
should we mention anywhere that this only works because we're not using controller-based privacy?

Reply:
Based on @alwa-nordic comments in https://github.com/zephyrproject-rtos/zephyr/issues/46395#issuecomment-1228233889 it does sound like RPA is fully controlled by the host when `CONFIG_BT_PRIVACY=y`

Reply:
Yeah that's true for now I guess. But there is no mention of this dependency (not BT_PRIVACY, but rather why it depends on host-based priv) either in the Kconfig or in the new callbacks' docs. 

I'll approve this PR since it's just changelog. @kapi-no could you add some disclaimer or a kconfig symbol `BT_PRIVACY_HOST` guarding these callbacks in a future PR?

Reply:
I think that the Zephyr Bluetooth stack works in the Host-based privacy mode when the BT_PRIVACY configuration is enabled. I guess there is no need to explicitly state that this particular feature works only for the Host-based privacy

Reply:
yeah it's this `generally works` that I don't like. It's hard enough to make sense of how privacy works re:the spec, but having implicit dependencies littered like this throughout the code just sets up a trap for the next guy.

`BT_PRIVACY` makes no mention whatsoever of the way it's implemented, we currently use host-based privacy for most stuff (like advertising with an rpa) and controller for other stuff (like scanning for RPAs).

So I'd really like for @alwa-nordic 's suggestion to add an assert (or kconfig check) to be implemented.